File: proje1/BilgiPaketi.java
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class BilgiPaketi{
    private List<Ders> tumDersler;
    public BilgiPaketi(){
        tumDersler=new ArrayList<>();
    }
    public void DersEkle(Ders ders){
           tumDersler.add(ders);
    }
    public void transkriptSorgula(ogrenci Ogrenci) {
        System.out.println(Ogrenci.getName() + " adlı öğrencinin transkripti:");
        
        // Öğrencinin tüm ders ve notlarını almak
        Map<Ders, double[]> ogrenciDersNotlari = Ogrenci.getDersNotlari();
        
        for (Ders ders : tumDersler) {
            // Eğer öğrenci bu derse kayıtlıysa ve notları varsa
            if (ogrenciDersNotlari.containsKey(ders)) {
                double[] notlar = ogrenciDersNotlari.get(ders);  // Öğrencinin derse ait notları
                String harfNotu = ders.notHesapla(notlar);       // Dersin not hesaplama fonksiyonu kullanılarak harf notu hesaplanır
    
                System.out.println("Ders: " + ders.getDersAdi());
                System.out.println("Dersin Kredisi: " + ders.kredi);
                System.out.println("Harf Notu: " + harfNotu);
            }
        }
    }
    
}

File: proje1python/BilgiPaketi.py
class BilgiPaketi:
    def __init__(self):
        self.tum_dersler = []  

    def ders_ekle(self, ders):
        self.tum_dersler.append(ders)  

    def transkript_sorgula(self, ogrenci):
        print(f"{ogrenci.get_name()} adli ogrencinin transkripti:")

        ogrenci_ders_notlari = ogrenci.get_ders_notlari()

        for ders in self.tum_dersler:
            
            if ders in ogrenci_ders_notlari:
                notlar = ogrenci_ders_notlari[ders]  
                harf_notu = ders.not_hesapla(*notlar)  

                print(f"Ders: {ders.get_ders_adi()}")
                print(f"Dersin Kredisi: {ders.kredi}")
                print(f"Harf Notu: {harf_notu}")
